### PR TITLE
PHP: remove invalid include-file and adjust include-file sort

### PR DESCRIPTION
**Pull request by thinkerou**
The pr remove some invalid or duplicate include files and adjust these files sort:
- first part is the correspond include file of source file,
- second part is php standard include file,
- third part is grpc include file,
- last part is some include files on php-grpc ext directory.

Please @stanley-cheung review the pr, thanks!

### CODE REVIEW

**Comment by grpc-testing** [comment]
Thanks for your pull request. The automated tests will run as soon as one of the admins verifies this change is ok for us to run on our infrastructure.

**Comment by grpc-testing** [comment]
Thanks for your pull request. The automated tests will run as soon as one of the admins verifies this change is ok for us to run on our infrastructure.

**Comment by grpc-testing** [comment]
Thanks for your pull request. The automated tests will run as soon as one of the admins verifies this change is ok for us to run on our infrastructure.

**Comment by grpc-testing** [comment]
Thanks for your pull request. The automated tests will run as soon as one of the admins verifies this change is ok for us to run on our infrastructure.

**Comment by grpc-testing** [comment]
Thanks for your pull request. The automated tests will run as soon as one of the admins verifies this change is ok for us to run on our infrastructure.

**Comment by stanley-cheung** [comment]
Ok to test

**Comment by stanley-cheung** [comment]
Compile error
```
/var/local/git/grpc/src/php/ext/grpc/byte_buffer.c  -fPIC -DPIC -o .libs/byte_buffer.o
In file included from /usr/include/php5/main/php.h:34:0,
                 from /var/local/git/grpc/src/php/ext/grpc/byte_buffer.c:21:
/usr/include/php5/main/php_config.h:2487:0: error: "_GNU_SOURCE" redefined [-Werror]
 #define _GNU_SOURCE 1
 ^
In file included from /var/local/git/grpc/include/grpc/impl/codegen/grpc_types.h:22:0,
                 from /var/local/git/grpc/include/grpc/byte_buffer.h:22,
                 from /var/local/git/grpc/include/grpc/grpc.h:24,
                 from /var/local/git/grpc/src/php/ext/grpc/byte_buffer.h:22,
                 from /var/local/git/grpc/src/php/ext/grpc/byte_buffer.c:19:
/var/local/git/grpc/include/grpc/impl/codegen/port_platform.h:153:0: note: this is the location of the previous definition
 #define _GNU_SOURCE
 ^
cc1: error: unrecognized command line option "-Wno-parentheses-equality" [-Werror]
cc1: all warnings being treated as errors
Makefile:181: recipe for target 'byte_buffer.lo' failed
```

**Comment by thinkerou** [comment]
OK, I will fix it later.

**Comment by stanley-cheung** [comment]
Still the same error

**Comment by thinkerou** [comment]
Hi, @stanley-cheung `Basic Tests Linux (opt)` and `Basic Tests Linux (dbg)` have fixed, the fixed method is that add macro in `byte_buffer.c`, like this:

    #ifdef _GNU_SOURCE
      #undef _GNU_SOURCE
    #endif

But I do not know whether the method is best method.

File: src/php/ext/grpc/byte_buffer.c [hunk 1]
```diff
@@ -16,23 +16,15 @@
  *
  */
 
-#ifdef HAVE_CONFIG_H
-#include "config.h"
+#include "byte_buffer.h"
+
+#ifdef _GNU_SOURCE
```
Comment (thinkerou) [inline src/php/ext/grpc/byte_buffer.c]: Fix compile error #12360#issuecomment-326484589

File: src/php/ext/grpc/php_grpc.h [hunk 2]
```diff
@@ -20,8 +20,21 @@
 #ifndef PHP_GRPC_H
 #define PHP_GRPC_H
 
+#ifdef HAVE_CONFIG_H
+#include "config.h"
+#endif
+
 #include <stdbool.h>
 
+#include <php.h>
```
Comment (thinkerou) [inline src/php/ext/grpc/php_grpc.h]: the php standard include files should be in the front.

File: src/php/ext/grpc/byte_buffer.c (hunk 1 above)
Comment (stanley-cheung) [reply src/php/ext/grpc/byte_buffer.c]: I don't particular like this solution. Can we understand the root cause better?

File: src/php/ext/grpc/byte_buffer.c [hunk 3]
```diff
@@ -16,23 +16,15 @@
  *
  */
 
-#ifdef HAVE_CONFIG_H
-#include "config.h"
+#include "byte_buffer.h"
```
Comment (stanley-cheung) [inline src/php/ext/grpc/byte_buffer.c]: Can you just move this line L19 to below L25? Then you don't need the whole block of
```
#ifdef _GNU_SOURCE
  #undef _GNU_SOURCE
#endif
```

Comment (thinkerou) [reply src/php/ext/grpc/byte_buffer.c]: sooo, the block codes have in php lib, thanks a lot, fixed.

File: src/php/ext/grpc/byte_buffer.c (hunk 1 above)
Comment (thinkerou) [reply src/php/ext/grpc/byte_buffer.c]: done.

**Comment by ZhouyihaiDing** [comment]
Can you please rebase the PR?
Recently there are some changes about the persistent connection. Since the test for the the persistent channel under php-fpm mpde is still under working, I have to test it manually. I am appreciated if you can rebase the PR so I can patch and test it easier. Thanks!

**Comment by thinkerou** [comment]
Sorry, the rebase have problem, solve it later on.

**Comment by thinkerou** [comment]
sorry, rebase and squash error, please see #15373 , thanks a lot!

**Pull request closed without merging**